src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 12;
+    private static final double GROW_COEFFICIENT = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity < 0) {
+            throw new IllegalArgumentException("Illegal Capacity: "
+                    + initialCapacity);
+        }
+        elementData = (T[]) new Object[initialCapacity > DEFAULT_CAPACITY
+                ? initialCapacity : DEFAULT_CAPACITY];
+    }
```

Comment:
```suggestion
        if (initialCapacity <= 0) {
            throw new IllegalArgumentException("Illegal Capacity: "
                    + initialCapacity);
        }
        elementData = (T[]) new Object[initialCapacity];
    }
```

I think we can not limit the size of the array by DEFAULT_CAPACITY in this case